firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -221,8 +226,31 @@ public void onFailure(@NonNull Exception e) {
    *     is cached from checkForUpdate
    */
   @NonNull
-  public UpdateTask updateApp() {
-    return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
+  public UpdateTask updateApp() throws FirebaseAppDistributionException {
+
+    if (updateAppTaskCompletionSource != null
+        && !updateAppTaskCompletionSource.getTask().isComplete()) {
+      updateAppCancellationSource.cancel();
+    }
+
+    updateAppCancellationSource = new CancellationTokenSource();
+    updateAppTaskCompletionSource =
+        new TaskCompletionSource<>(updateAppCancellationSource.getToken());
+    Context context = firebaseApp.getApplicationContext();
+
+    if (appDistributionReleaseInternal == null) {
+      throw new FirebaseAppDistributionException(
```

Comment:
Since context is only used here you can inline `firebaseApp.getApplicationContext().getString` here

Reply:
I thought that maybe I would need it again when implementing the APK part so I left it there for now.  is it better to inline it now and change it back if need be, or inline it later if it ends up not being needed again?

Reply:
Ahh okay. Its not a big deal. You can leave this as is.